karate-tests/src/test/java/patients/patientAccess/updatePatient.feature

```diff
@@ -81,3 +81,134 @@ Feature: Patient updates their details
+      * match response == read('classpath:mocks/stubs/errorResponses/INVALID_UPDATE.json')     
+  
+
+  Scenario: Patient can update their emergency contact details and place of birth
+    * def accessToken = karate.call('classpath:auth/auth-redirect.feature', {userID: p9number, scope: 'nhs-login'}).accessToken
+    * def requestHeaders = call read('classpath:auth/auth-headers.js')
+    * configure headers = requestHeaders
+    * path 'Patient', p9number
+    * method get
+    * status 200
+    * def originalVersion = parseInt(response.meta.versionId)
+    * def originalEtag = karate.response.header('etag')
+    * def placeOfBirthDls = read('classpath:patients/requestDetails/add/placeOfBirth.json')
+    * def placeOBirthUrl = placeOfBirthDls.patches[0].value.url
+    * def originalTelecom = response.contact[0].telecom[0].value
+    * def contactID = response.contact[0].id
+    * def relationshipDetails =  response.contact[0].relationship
+    * def pobDetails = response.extension.find(x => x.url == placeOBirthUrl)
+    * def pobDistrict = pobDetails.valueAddress.district
+    * def pobcountry = pobDetails.valueAddress.country
+    * def pobIndex = response.extension.findIndex(x => x.url == placeOBirthUrl)
+    * def pobPath =  "/extension/" + pobIndex
+
+    # update emergency contact details
+    * configure headers = call read('classpath:auth/auth-headers.js') 
+    * header Content-Type = "application/json-patch+json"
+    * header If-Match = originalEtag
+    * def newMobileNumber = faker.phoneNumber()
+    * path 'Patient', p9number
+    * def requestbody =
+    """
+    {
+      "patches":[
+        {"op":"replace",
+        "path":"/contact/0",
+        "value":{
+          "id": "#(contactID)",
+          "relationship":"#(relationshipDetails)",
+          "telecom":[
+            {"system":"phone",
+            "value":"#(newMobileNumber)"}
+            ]}}]}
+     """ 
+    * request requestbody      
+    * method patch
+    * status 200
+    * match response.contact[0].id contains contactID
+    * match response.contact[0].telecom[0].value == newMobileNumber
+    * match parseInt(response.meta.versionId) == originalVersion + 1
+
+    #update place of birth details
+    * def cityName = faker.cityName()
+    * configure headers = call read('classpath:auth/auth-headers.js') 
+    * header Content-Type = "application/json-patch+json"
+    * header If-Match = karate.response.header('etag')
+    * path 'Patient', p9number
+    * request 
+    """
+    {
+      "patches":[
+        {"op":"replace",
+        "path":"#(pobPath)",
+        "value":{
+          "url": "#(placeOBirthUrl)",
+        "valueAddress": {
+                "city": "#(cityName)",
+                "district": "#(pobDistrict)",
+                "country": "#(pobcountry)"
+            }
+        }}]}
+     """  
+    * method patch
+    * status 200  
+    * def updatedPobCity = response.extension[pobIndex].valueAddress.city
+    * match updatedPobCity == cityName
+
```

Comment:
Possibly add another response.meta.versionId check here

Reply:
Thank you, added response.meta.versionId check